datasource/example/datasource.go

```diff
@@ -22,21 +22,21 @@ func (d *MyDatasource) Query(ctx context.Context, tr datasource.TimeRange, ds da
 			RefID: "A",
 			DataFrames: []*dataframe.Frame{
 				dataframe.New("http_requests_total", dataframe.Labels{"service": "auth", "env": "prod"},
-					dataframe.NewField("timestamp", dataframe.FieldTypeTime, []time.Time{time.Now(), time.Now(), time.Now()}),
-					dataframe.NewField("value", dataframe.FieldTypeNumber, []float64{45.0, 49.0, 29.0}),
+					dataframe.NewField("timestamp", []time.Time{time.Now(), time.Now(), time.Now()}),
```

Comment:
A future improvement that would be nice for plugin developers is if there instead was `dataframe.NewTimeField`, `dataframe.NewNullableTimeField` etc exposed to get compile time type checks.